src/global_scheduler/global_scheduler.c

```diff
@@ -41,6 +51,14 @@ void assign_task_to_local_scheduler(global_scheduler_state *state,
       get_local_scheduler(state, local_scheduler_id);
   local_scheduler->num_tasks_sent += 1;
   local_scheduler->num_recent_tasks_sent += 1;
+  /** Resource accounting update for this local scheduler. */
+  for (int i = 0; i < MAX_RESOURCE_INDEX; i++) {
+    /** Subtract task's resource from the cached dynamic resource capacity for
+     *  this local scheduler. This will be overwritten on the next heartbeat. */
+    local_scheduler->info.dynamic_resources[i] =
+        MAX(0, local_scheduler->info.dynamic_resources[i] -
```

Comment:
We shouldn't take a MAX. It should be allowed to go negative (otherwise, once all nodes hit 0 the global scheduler won't know how to distinguish between them even if way more tasks have been assigned to one of them).